src/resources/cpp/lambdas.md

```diff
@@ -0,0 +1,365 @@
+
+Different captures can be mixed together:
+
+```cpp
+int a = 1;
+int b = 2;
+int c = 3;
+
+auto foo = [=, &b, d = c]() {
+	// `a` is implicitly captured by value,
+	// `b` is explicitly captured by reference, and
+	// `d` is initialized to `c`.
+	return a + b + d;
+};
+```
+
+However, `[=]` may only be followed by captures by reference and `[&]` may only be followed by captures by value.
+
+Implicit captures can be nested multiple times:
+
+```cpp
+int x = 0;
+
+[&]() {
+	[&]() {
+		x = 5;
+	}();
+}();
+```
+
+&nbsp;
+
+### Parameters
+
+Lambda parameters work the same way as normal function parameters, and `auto` parameters make a lambda's `operator()`
+implicitly templated.
+
+If a lambda takes no parameters, the parameter list may be omitted entirely:
+
+```cpp
+auto very_important_number = [] { return 4; };
+```
+
+Finally, the explicit `this` parameter can also be in lambdas since C++23:
+
+```cpp
+auto fibonacci = [](this auto self, int n) {
+	if (n < 2) return n;
+	return self(n - 1) + self(n - 2);
+};
+```
+
+&nbsp;
+
+### Return type
+
+A lambda's return type may be specified with an arrow:
+
+```cpp
+auto add = [](int x, int y) -> int { return x + y; };
+```
+
+Omitting the return type is the same as writing `-> auto`.
+
+&nbsp;
+
+### Specifiers
+
+- `constexpr` - Explicitly specifies that a lambda's `operator()` is a
+  [constexpr function](https://en.cppreference.com/w/cpp/language/constexpr#constexpr_function).
+  - Mutually exclusive with `consteval`.
+  - Lambdas are implicitly marked `constexpr`, if possible.
+- `consteval` - Makes a lambda's `operator()` an
+  [immediate function](https://en.cppreference.com/w/cpp/language/consteval).
+  - Mutually exclusive with `constexpr`.
+- `static` - Makes a lambda's `operator()` a
+  [static member function](https://en.cppreference.com/w/cpp/language/static#Static_member_functions).
+  - Mutually exclusive with `mutable`.
+  - Cannot be used if the captures list is not empty, or an explicit `this` parameter is present.
+- `mutable` - Allows the body of the lambda to modify the variables captured by value.
+  - Mutually exclusive with `static`.
+  - Cannot be used if an explicit `this` parameter is present.
+
+```cpp
+auto next = [i = 0] mutable { return i++; };
+
+next() // 0
+next() // 1
+next() // 2
+```
+
+These may be followed by a `noexcept` specifier, to determine whether calling the lambda may throw an exception.
+
+&nbsp;
+
+### Template parameters
+
+A lambda's `operator()` may be templated to accept template parameters since C++20:
+
+```cpp
+auto lambda = []<typename T>(const T& x) {
+	return x;
+};
+
+// Deduce template argument from function argument
+lambda(5);
+
+// Pass template argument explicitly
+lambda.template operator()<double>(5);
+```
+
+&nbsp;
+
+### Attributes
+
+Lambdas may be given attributes that apply to their `operator()`s since C++23:
+
+```cpp
+auto very_important_number = [][[nodiscard]] { return 4; };
```

Comment:
Very minor style nit but I'd suggest putting a space between the attribute, and the capture block, otherwise it's just a big block of square brackets `[][[`

```suggestion
auto very_important_number = [] [[nodiscard]] { return 4; };
```